src/drawing/backend_impl/canvas.rs

```diff
@@ -77,8 +84,11 @@ impl DrawingBackend for CanvasBackend {
         let window = window().unwrap();
         let mut dpr = window.device_pixel_ratio();
         dpr = if dpr == 0.0 { 1.0 } else { dpr };
-        ((self.canvas.width() as f64 / dpr) as u32,
-         (self.canvas.height() as f64 / dpr) as u32)
+        let rect = self.canvas.get_bounding_client_rect();
```

Comment:
Thanks for point this out, but I believe this is an actually wrong fix to another issue. 

Basically, the invariant here we want to maintain is in: If plotters want to draw a pixel at (width, height) it should present in the right bottom, that is the right positioning. As long as the CanvasBackend dynamically get the logical size of the canvas. So this part shouldn't related to DPR at all. 

Now I just realized this is a really complicated story:

I previously merged a PR which does wrong scale adjustment on DPR (in the WASM example's JS code). I found the similar method is described [here](https://www.html5rocks.com/en/tutorials/canvas/hidpi/) but the whole point is the the blog post assume the drawing code hard coded the canvas size, thus once we adjust the canvas logic size, we should then scale it, so that we don't need to modify the code. But Plotters does a completely different thing, the backend check the current logic size each time and all other part of the crate uses this. But it looks Ok on normal DPI screen.


After that somebody else realize that on a highDPI screen it has a problem, which comes up with this "fix". Which divide the DPR so that this can cancel the previously wrong scale out. But this is not really a fix.

Thanks for point this out and I am going to clean up the mess soon. If you don't mind could you please change this to 
```
(self.canvas.width(), self.canvas.height())
```
This is what it should be